views/application.py

```diff
@@ -93,3 +94,50 @@ def update(self, request, *args, **kwargs):
                 list(set(new_team_member_ids)-set(old_team_member_ids))
             )
         return response.Response(serializer.data)
+
+
+class ApplicationHiddenDetailView(generics.GenericAPIView):
+    """
+    View details of Application which should not be exposed directly
+    """
+
+    permission_classes = [permissions.IsAuthenticated, ]
+
+    def post(self, request, *args, **kwargs):
+        """
+        Authenticate if the requesting user is Team Member
+        of the application and check the password entered.
+        :param request: the request being processed
+        :param args: arguments
+        :param kwargs: keyword arguments
+        :return: response containing client_secret
+        """
+
+        try:
+            app = Application.objects.get(pk=request.data.get('id'))
+
+            if app.team_members.filter(id=request.person.id).exists():
```

Comment:
Don't extract data from request in function calls. Extract person first and then store it in person. Get its id in other variable say person_id then use that here. Even for getting the password, once try making this request without sending the password parameter in the request object and you will see how this will throw a 500. Being an open source project we should really care about the edge cases. Perfection is much more desirable then completion.